### PR TITLE
Chore: handle disable-line comments in a separate pass

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other, please explain:

**What changes did you make? (Give an overview)**

This is another patch on https://github.com/eslint/eslint/pull/9322.

It makes the fix simpler again by just treating `line` and `block` directive comments separately, to avoid needing to keeping track of extra state.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular

### CODE REVIEW

File: tests/lib/util/apply-disable-directives.js [hunk 1]
```diff
@@ -371,19 +388,6 @@ describe("apply-disable-directives", () => {
                 []
             );
         });
-
-        it("keeps problems on the next line if there is an eslint-enable comment before the problem on the next line", () => {
-            assert.deepEqual(
-                applyDisableDirectives({
```
Comment (gyandeeps) [inline tests/lib/util/apply-disable-directives.js]: why did you remove this test?

**Comment by gyandeeps** [comment]
Overall I like the approach.. much cleaner.... but now we are iterating twice (not saying its an issue)...

File: tests/lib/util/apply-disable-directives.js (hunk 1 above)
Comment (not-an-aardvark) [reply tests/lib/util/apply-disable-directives.js]: If we're handling `eslint-disable` and `eslint-enable` comments separately from `eslint-disable-line` and `eslint-disable-next-line`, then it doesn't make sense for `eslint-enable` comments to affect the next line.

That test was added as part of https://github.com/eslint/eslint/commit/a32ec3620bf9aa8fae0c7ae94163f3e4a732b3ce because it was the previous behavior before the refactor, but I don't know of anyone relying on it (and it's undocumented anyway).

Comment (gyandeeps) [reply tests/lib/util/apply-disable-directives.js]: This test gave me some hard time too.
But I think the behavior defined inside this test does make sense. I dont think we should remove it.
